### PR TITLE
Fixes #14306,#15062 - Create an upgrade check script

### PR DESCRIPTION
**Pull request by daviddavis**
Will update the upgrade docs on katello.org after merging this.

### CODE REVIEW

**Comment by beav** [comment]
[test]

**Comment by jlsherrill** [comment]
tests won't pass on the 2.4 branch, jenkins isn't smart enough :(

File: lib/katello/tasks/upgrade_check.rake [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    task_count = ::ForemanTasks::Task.active.where("label != '#{CP_LISTEN_ACTION}'").count
+    task_status = task_count > 1 ? fail : success
+    puts "Checking for running tasks..."
+    puts "[#{task_status}] - There are #{task_count} active tasks.\n\n"
+
+    # check the current version
+    version_status = current_version == DESIRED_VERSION ? success : fail
+    puts "Checking the current version..."
+    puts "[#{version_status}] - Current version of #{SYSTEM} is #{current_version} " \
+      "and needs to be #{DESIRED_VERSION}\n\n"
```
Comment (stbenjam) [inline lib/katello/tasks/upgrade_check.rake]: Hmm, is this supposed to solve #14306?  It won't exist on < 2.4.2 so users will always be running at least that, and if they already upgraded to 3.0 there's not much we can do about it.

File: lib/katello/tasks/upgrade_check.rake [hunk 2]
```diff
@@ -0,0 +1,36 @@
+DESIRED_VERSION = "2.4.2"
+SYSTEM = "Katello"
```
Comment (stbenjam) [inline lib/katello/tasks/upgrade_check.rake]: Is the idea Satellite could override this? We're not consistently using it then, e.g. Katello 3.0 appears in the strings below

File: lib/katello/tasks/upgrade_check.rake (hunk 1 above)
Comment (daviddavis) [inline lib/katello/tasks/upgrade_check.rake]: You're right although I guess it wouldn't hurt to leave it? Or were you proposing something else?

Comment (stbenjam) [inline lib/katello/tasks/upgrade_check.rake]: Nah, I spent a while thinking about this and tried a few different things but could never come up with a way to solve it how they requested in the BZ.  You can leave it, I suppose.  It lets the user know when we expected them to run the script I guess.

**Comment by stbenjam** [comment]
Otherwise looks ok to me, APJ

**Pull request merged**